src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 11 May 2020
+
+@author: Philip Winkler
+
+Copyright © 2020 Philip Winkler, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from odemis import model
+import logging
+import serial
+from pymodbus.client.sync import ModbusSerialClient
+import numpy
+import serial.tools.list_ports
+
+# Parameters for connection
+BAUDRATE = 230400
+TIMEOUT = 1
+BYTESIZE = 8
+PARITY = serial.PARITY_NONE
+STOPBITS = 1
+
+# Modbus level addresses
+SLAVE_UNIT = 2
+
+# Modbus registers
+BEAMDEFL_LX = 0  # lower x beam deflection element control
+BEAMDEFL_LY = 1  # lower y beam deflection element control
+BEAMDEFL_UX = 2  # upper x beam deflection element control
+BEAMDEFL_UY = 3  # upper y beam deflection element control
+
+RANGE_AMPS = [-42.2e3, 42.2e3]
+
+class BeamController(model.HwComponent):
+    """
+    Driver for the Thermofischer beam controller.
+    This class provides the .shift VA containing a tuple of two floats which describe
+    the x and y beam offset in m.
+    """
+
+    def __init__(self, name, role, amps_per_m, zero_offset_amps,
+                 rs485_serial_num, **kwargs):
+        """
+        :param port (str): port (e.g. /dev/ttyUSB0)
+        :param amps_per_m (float): conversion factor amplitude/meter
+        :param zero_offset_amps (float): amplitude for which beam deflection is 0
+        :param rs485_serial_num (str): serial number of RS485 adapter
+        """
+        # .hwVersion, .swVersion not available
+        model.HwComponent.__init__(self, name, role, **kwargs)
+
+        self._amps_per_m = amps_per_m
+        self._zero_offset_amps = zero_offset_amps
+
+        # Find port by RS485 adapter serial number
+        for p in serial.tools.list_ports.comports():
+            if rs485_serial_num in p.hwid:
+                # Found it!
+                port = p
+                break
+        else:
+            raise IOError("RS484 adapter with serial number %s not found!" % rs485_serial_num)
```

Comment:
This should be actually a HwError, to indicate to the backend, that it should try again.
Concerning the message string. RS484 -> RS485. Also, remove the "!", as it's not that much worrying for the user.